Auto (balance in charge station)

stoopipoopy opened the pull request.

stoopipoopy committed:
started autonomous and gyro thingie

stoopipoopy committed:
thing for the gyro thingy (you know what it is)

stoopipoopy committed:
Started adding gyro; need to change the
run command to a selector thingy

stoopipoopy committed:
finished balance command and added to auto

stoopipoopy committed:
added gyro id as constant

stoopipoopy committed:
Merge branch 'master' into Auto

Lyssia-Seiden commented on src/main/java/frc/robot/Commands/Autonomous.java:
```diff
@@ -0,0 +1,39 @@
+/** Add your docs here. */
+public class Autonomous {
+    SwerveSubsystem swerveSubsystem;
+
```
You may want to have a LoggedDashboardChooser here to select between different autos

Lyssia-Seiden commented on src/main/java/frc/robot/Commands/Autonomous.java:
```diff
@@ -0,0 +1,39 @@
+         */
+
+         return Commands.sequence(
+            swerveSubsystem.drive(() -> 2, () -> 0, () -> 0, false).withTimeout(2),
```
check the sign of this movement

Lyssia-Seiden commented on src/main/java/frc/robot/Subsystems/DrivebaseSubsystem/GyroModuleIO.java:
```diff
@@ -0,0 +1,26 @@
+        public double yaw = 0.0;
+        public double roll = 0.0;
+    }
+
```
You shouldnt have getters in an io interface, because the values passed through them are not logged. instead use values captured into the ioinputs instance in the subsystem

Lyssia-Seiden replied:
i made this mistake on our swerve code at chezy lmao it was not fun to realize all our logs were iffy because of that

Lyssia-Seiden commented on src/main/java/frc/robot/Subsystems/DrivebaseSubsystem/GyroModuleIO.java:
```diff
@@ -0,0 +1,26 @@
+        public double yaw = 0.0;
+        public double roll = 0.0;
+    }
+
```
you need an updateinputs method

Lyssia-Seiden commented on src/main/java/frc/robot/Subsystems/DrivebaseSubsystem/GyroModuleIOReal.java:
```diff
@@ -0,0 +1,55 @@
+    }
+    @Override
+    public double getPitch() {
+        
```
add updateInputs

Lyssia-Seiden commented on src/main/java/frc/robot/Subsystems/DrivebaseSubsystem/SwerveSubsystem.java:
```diff
@@ -128,6 +139,30 @@ public CommandBase drive(DoubleSupplier forward, DoubleSupplier side, DoubleSupp
+            drive(() -> 0,() -> 0,() -> 0, true), 
+            Commands.either(
+                drive(() -> 0,() -> 0,() -> 0,true),
+                  Commands.either(
```
could add some comments here instead of above it

stoopipoopy committed:
aded comments for balance command and change signs
fixed gyro io inputs to update itself and no getters
added logged dashboard chooser

stoopipoopy committed:
Merge branch 'Auto' of https://github.com/stoopipoopy/8033OffSeasonRobot into Auto

stoopipoopy committed:
final commit probably

Lyssia-Seiden commented on src/main/java/frc/robot/Commands/Autonomous.java:
```diff
@@ -0,0 +1,49 @@
+public class Autonomous {
+    
+    SwerveSubsystem swerveSubsystem;
+    LoggedDashboardChooser<Command> autoSystemType =
```
might want to rename
